src/main/java/com/dev/cinema/dao/impl/GenericDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.GenericDao;
+import com.dev.cinema.exceptions.DataProcessingException;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
```

Comment:
does it make sense to put this annotation on abstract class, can we create instance of it?